net/Socket.cpp

```diff
@@ -607,6 +609,21 @@ int SocketPoll::poll(int64_t timeoutMaxMicroS, bool justPoll)
             }
         }
 
+        if (pendingTransfers.size() > 0)
+            LOGA_TRC(Socket, "Invoking " << pendingTransfers.size() << " transfers");
+        for (const auto& pendingTransfer : pendingTransfers)
+        {
+            try
+            {
+                transfer(pendingTransfer);
+            }
+            catch (const std::exception& exc)
+            {
+                LOG_ERR("Exception while invoking poll [" << _name <<
+                        "] transfer: " << exc.what());
+            }
+        }
+
```

Comment:
I wonder if we should cleanup/free both the callbacks and the pendingTransfers here - to get simpler visibility into the lifecycle of the variables captured in their closures in various cases (?)

Reply:
https://github.com/CollaboraOnline/online/pull/11755 for that